src/components/modals/RequestFioAddressModal.js

```diff
@@ -0,0 +1,222 @@
+
+type AddressModalProps = {
+  onDone: ({ fioAddress: string, memo: string } | null) => mixed,
+  fioPlugin: EdgeCurrencyConfig,
+  isConnected: boolean
+}
+
+type AddressModalState = {
+  clipboard: string,
+  address: string,
+  addressError: string,
+  memo: string,
+  memoError: string,
+  addressFocus: boolean,
+  memoFocus: boolean,
+  addressValidationLoading: boolean
+}
+export class FioAddressModal extends Component<AddressModalProps, AddressModalState> {
+  constructor (props: AddressModalProps) {
+    super(props)
+    this.state = {
+      clipboard: '',
+      address: '',
+      addressError: '',
+      memo: '',
+      memoError: '',
+      addressFocus: true,
+      memoFocus: false,
+      addressValidationLoading: false
+    }
+  }
+
+  componentDidMount () {
+    this._setClipboard()
+  }
+
+  _setClipboard = async () => {
+    try {
+      const address = await Clipboard.getString()
+      const isFioAddress = await this.props.fioPlugin.otherMethods.isFioAddressValid(address)
+      if (!isFioAddress) return
+
+      this.setState({
+        clipboard: address
+      })
+    } catch (e) {
+      //
+    }
+  }
+
+  updateAddress = (address: string) => {
+    this.setState({
+      address
+    })
+  }
+
+  updateMemo = (memo: string) => {
+    this.setState({
+      memo
+    })
+  }
+
+  onPasteFromClipboard = () => {
+    const { clipboard } = this.state
+    this.updateAddress(clipboard)
+  }
+
+  onCancel = () => {
+    this.props.onDone(null)
+  }
+
+  onSubmit = async () => {
+    const validAddress = await this.validateAddress()
+    const validMemo = this.validateMemo()
+    if (validAddress && validMemo) {
+      this.props.onDone({ fioAddress: this.state.address, memo: this.state.memo })
+    }
+  }
+
+  onValidateAddress = (): void => {
+    this.validateAddress()
+  }
+
+  onValidateMemo = (): void => {
+    this.validateMemo()
+  }
+
+  validateAddress = async (): Promise<boolean> => {
+    const address: string = this.state.address
+    let accountIsAvailable
+    this.setState({
+      addressValidationLoading: true
+    })
+    try {
+      accountIsAvailable = await this.props.fioPlugin.otherMethods.isAccountAvailable(address)
+    } catch (e) {
+      accountIsAvailable = false
+    }
+    if (address && accountIsAvailable) {
+      this.setState({
+        addressError: '',
+        addressFocus: false,
+        memoFocus: true,
+        addressValidationLoading: false
+      })
+      return true
+    } else {
+      this.setState({
+        addressError: s.strings.fragment_send_send_to_fio_error_inline,
+        addressFocus: true,
+        memoFocus: false,
+        addressValidationLoading: false
+      })
+      return false
+    }
+  }
+
+  validateMemo = (): boolean => {
+    const memo: string = this.state.memo
+    if (!memo || (this.isASCII(memo) && memo.length <= 64)) {
+      this.setState({
+        memoError: ''
+      })
+      return true
+    } else {
+      this.setState({
```

Comment:
Missing proper error msg if memo is not ASCII "Invalid character in memo"

Reply:
added to memo error message in translation

Reply:
Please create 2 separate error messages. One for 64 char max and one for ASCII. use "Invalid character in memo" for ASCII error.

Reply:
created